### PR TITLE
Added Powered CP (CP @ maxed stardust)

### PR DESCRIPTION
**Pull request by marksterxxx**
-Modified Console display for pokemon info

### CODE REVIEW

**Comment by marksterxxx** [comment]
untitled

Just a suggestion, 
I think we can have PercentCP = currentCP/PoweredCP percentage
and put weight on PecentCP and Perfection, say 0.5 for each(depending on setting), 
then we can compute a Battle Rating = 0.5*PercentCP + 0.5*Perfection as basis for comparing pokemon. That way, those who want pure IV can just 100% weight for Perdection, while others who want to quantity can add more weight to PercentCP

**Comment by DurtyFree** [comment]
Do we really need this? Need opinions.

**Comment by ModernTimes** [comment]
These information are essential for powergamers. So yes, make it so!
(Maybe add an option to turn the details on/off.)

**Comment by forReason** [comment]
uhhhm. what does the powered stand for?
maybe a bit more precise naming would be great

File: PoGo.PokeMobBot.Logic/Tasks/DisplayPokemonStatsTask.cs [hunk 1]
```diff
@@ -22,48 +22,48 @@ public class DisplayPokemonStatsTask
...
+            var trainerLevel = 30;
             var highestsPokemonCp =
                 await session.Inventory.GetHighestsCp(session.LogicSettings.AmountOfPokemonToDisplayOnStart);
             var highestsPokemonCpForUpgrade = await session.Inventory.GetHighestsCp(50);
             var highestsPokemonIvForUpgrade = await session.Inventory.GetHighestsPerfect(50);
             var pokemonPairedWithStatsCp =
                 highestsPokemonCp.Select(
                     pokemon =>
-                        Tuple.Create(pokemon, PokemonInfo.CalculateMaxCp(pokemon),
+                        Tuple.Create(pokemon, Tuple.Create(PokemonInfo.CalculateMaxCp(pokemon), (int)PokemonInfo.GetMaxCpAtTrainerLevel(pokemon, trainerLevel)),
```
Comment (bmdr38) [inline PoGo.PokeMobBot.Logic/Tasks/DisplayPokemonStatsTask.cs]: Can we stop abusing Tuple?  If you need to store/retrieve information about something, create a class with the respected properties/fields that clearly identify their purpose.  Especially the fact that you have nested Tuple in Tuple.  This screams poor design.

Can you imagine now, trying to use this list?  You get something like: Item1, Item2, Item2.Item1, Item2.Item2, Item3, Item4, etc etc.

This is not clear at all in usages later.

**Comment by bmdr38** [comment]
Consider changing or overloading GetMaxCpAtTrainerLevel to always do the max (for level 40 + 1.5) so that you can really show the true maxed out stats instead of only the maxed out stats for your current level.

File: PoGo.PokeMobBot.Logic/Tasks/DisplayPokemonStatsTask.cs (hunk 1 above)
Comment (marksterxxx) [inline PoGo.PokeMobBot.Logic/Tasks/DisplayPokemonStatsTask.cs]: done...

**Comment by marksterxxx** [comment]
@bmdr38  I now set it to display max cp @ lvl 40 trainer.
@class for pokemon analysis now in Pokemoninfo.cs

**Comment by marksterxxx** [comment]
@DurtyFree  as perfection or the best cp can be used to compare same pokemon, powered cp can be used to compare different pokemon ex: I would still prefer 90%IV lvl 1 lapras than a 100%IV lvl 30 chansey because in the end, after powering up, my lapras would have 3k cp and chansey would only have 700 . This will help players prioritize which pokemon type to keep.

**Comment by bmdr38** [comment]
My only final concern would be to move the trainer level into the analysis class as a default, but I think I might be getting too picky :).  I hope to see this merged, both for the refactoring and the informational purposes of seeing the stats potentials of my pokemon

**Comment by DurtyFree** [comment]
@marksterxxx Can you add a setting to enable / disable the detailed informations for this? And please do a new pr, this one got quite huge.

**Comment by marksterxxx** [comment]
new pr

**Pull request closed without merging**